MapboxNavigation/NavigationMapView.swift

```diff
@@ -1362,13 +1368,17 @@ extension NavigationMapView {
      - parameter extrudesBuildings: Switch which allows to highlight buildings in either 2D or 3D. Defaults to true.
      */
     public func highlightBuildings(at coordinates: [CLLocationCoordinate2D], in3D extrudesBuildings: Bool = true) {
-        highlightBuildings(with: Set(coordinates.compactMap({ buildingIdentifier(at: $0) })), in3D: extrudesBuildings)
+        buildingHighlightPollingTimer?.invalidate()
```

Comment:
Since `SimulatedLocationManager` uses `updateInterval` of 1 second timer invalidation might cause inability to highlight building if polling timer timeout is higher than 1 second. This also applies to real-life use case when location manager is going to deliver location data. We might have to reconsider current implementation and think whether `NavigationMapView` is really best place to store polling timer.

Reply:
Agreed. In general, we should avoid storing any additional state on NavigationMapView because we aren’t full in control of the lifecycle of its contents.